track/track.py

```diff
@@ -332,28 +332,37 @@ def __repr__(self):
     def __len__(self):
         return len(self.bounds_history)
 
-    def get_best_human_track(track_meta, tag_precedence, min_confidence=-1):
+    @classmethod
+    def get_best_human_tag(cls, track_meta, tag_precedence, min_confidence=-1):
         """ returns highest precidence non AI tag from the metadata """
 
-        track_tags = track_meta.get("TrackTags")
-        if not track_tags:
-            return None
+        track_tags = track_meta.get("TrackTags", [])
         track_tags = [
             tag
             for tag in track_tags
             if not tag.get("automatic", False)
             and tag.get("confidence") > min_confidence
         ]
+
+        if not track_tags:
+            return None
+
         tag = None
-        if track_tags:
-            default_prec = tag_precedence.get("default", 100)
-            tag = min(
-                track_tags,
-                key=lambda tag: Track.tag_order(tag, tag_precedence, default_prec),
-            )
+        default_prec = tag_precedence.get("default", 100)
+        best = None
+        for track_tag in track_tags:
+            ranking = cls.tag_ranking(track_tag, tag_precedence, default_prec)
+
+            # if 2 track_tags have same confidence ignore both
+            if ranking == best:
+                tag = None
```

Comment:
does this mean a lower confidence tag could be returned if 2 other high confidence tags exist? 

For example if a track's tags were:
- cat 0.8
- possum 0.8
- rodent 0.5

... I wouldn't want "rodent" to be returned as the best tag. Tracks tagged like this should probably just be ignored for training purposes.

Reply:
should still return None as the actual best_value isn't ever reset

Reply:
Right I see, because only the tag is reset, but not `ranking`